<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to SJD (Structured Joint Decomposition) • SJD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to SJD (Structured Joint Decomposition)">
<meta property="og:description" content="SJD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SJD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/SJD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SJD_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to SJD (Structured Joint Decomposition)</h1>
                        <h4 class="author">Huan Chen</h4>
                        <h4 class="author">Jinrui Liu</h4>
                        <h4 class="author">Carlo Colantuoni</h4>
            
            <h4 class="date">15 November, 2021</h4>
      
      
      <div class="hidden name"><code>SJD.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>SJD (Structured Joint Decomposiiton) is a R package for visualizing biologically structured gene expression matrix environment based on low rank models. Currently, it provides user 4 different styles of decomposition: (1) Separately, (2) Concatenately, (3) Jointly, (4) Sequentially.</p>
<p>This package implements four categories of algorithms to decompose multiple datasets, (1) Separately, (2) Concatenately, (3) Jointly, (4) Two Stage Sequentially.</p>
<p>For the first three categories, there are three available algorithms: Principal Component Analysis (PCA), Independent Component Analysis (ICA) and Nonnegative Matrix Factorization (NMF). For each method, the algorithm takes three arguments, <code>dataset</code>, <code>group</code> and <code>comp_num</code>, specifying which datasets to be used, what is the structure among the datasets and what’s the dimension for each component.</p>
<p>For the last category, there is one algorithm, called two-staged linked component analysis, which is a PCA based statistical model.</p>
</div>
<div id="install-and-load-sjd-package" class="section level1">
<h1 class="hasAnchor">
<a href="#install-and-load-sjd-package" class="anchor"></a>Install and load <code>SJD</code> package</h1>
<p>To install this package in R, run the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"CHuanSite/SJD"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SJD</span><span class="op">)</span></code></pre></div>
</div>
<div id="jointly-decompose-4-neurogenesis-genomics-data-with-sjd-package" class="section level1">
<h1 class="hasAnchor">
<a href="#jointly-decompose-4-neurogenesis-genomics-data-with-sjd-package" class="anchor"></a>Jointly decompose 4 NeuroGenesis genomics data with <code>SJD</code> package</h1>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h2>
<p>We load the structured neuro genesis data <code>NeuroGenesis4</code> and its information data <code>NeuroGenesis.info</code> contained in the package <code>SJD</code> into the environment.</p>
<p>There are four separate data sets in the <code>NeuroGenesis4</code> list:</p>
<ul>
<li>Meissner.inVitro.bulk.Hs</li>
<li>LIBD.AZ.inVitro.bulk.Hs</li>
<li>Geschwind.inVivo.sc.Hs</li>
<li>Jabaudon.inVivo.sc.Mm</li>
</ul>
<p>Note that the four data sets in <code>NeuroGenesis4</code> are sampled from the original large list of data sets, which is a very small subset of the real data. Exact the same process can be applied to the original data, please download the original data sets at <a href="https://drive.google.com/file/d/1jf4f_dxps47yT89m8Tkpviyci_XoOtGk/view?usp=sharing">data</a> and <a href="https://drive.google.com/file/d/1aGYAxZDYJKOLLO1LmFk9vPYUmtmS1TjZ/view?usp=sharing">info</a></p>
<p>This part of code is to use the full data sets and not evaluted in this vignette</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("googledrive")</span>
<span class="co"># </span>
<span class="co"># library(googledrive)</span>
<span class="co"># </span>
<span class="co"># ## Download NeuroGenesis4</span>
<span class="co"># drive_download(file = "https://drive.google.com/u/0/uc?export=download&amp;id=1jf4f_dxps47yT89m8Tkpviyci_XoOtGk", path = "./NeuroGenesis4.RData", overwrite = TRUE)</span>
<span class="co"># </span>
<span class="co"># ## Download NeruGenesis4.info</span>
<span class="co"># drive_download(file = "https://drive.google.com/file/d/1aGYAxZDYJKOLLO1LmFk9vPYUmtmS1TjZ/view?usp=sharing", path = "./NeuroGenesis4.info.RData", overwrite = TRUE)</span>
<span class="co"># </span>
<span class="co"># load(file = "./NeuroGenesis4.RData")</span>
<span class="co"># load(file = "./NeuroGenesis4.info.RData")</span></code></pre></div>
<p>In this vignette, we use the small sampled data sets instead, note that the whole procedure can be implemented without chaning anything by just using the full data sets and info.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"NeuroGenesis4"</span>, package <span class="op">=</span> <span class="st">"SJD"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"NeuroGenesis4.info"</span>, package <span class="op">=</span> <span class="st">"SJD"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">NeuroGenesis4</span><span class="op">)</span>
<span class="co">#&gt; List of 4</span>
<span class="co">#&gt;  $ Meissner.inVitro.bulk.Hs: num [1:40, 1:15] 11.9 11.4 11.5 11.4 11.7 ...</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   .. ..$ : chr [1:40] "ACTB" "ACTG1" "RPS11" "RPS27" ...</span>
<span class="co">#&gt;   .. ..$ : chr [1:15] "ESC.1" "ESC.2" "NE.1" "NE.2" ...</span>
<span class="co">#&gt;  $ LIBD.AZ.inVitro.bulk.Hs : num [1:40, 1:15] 9.8 9.91 9.93 10.11 9.9 ...</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   .. ..$ : chr [1:40] "ENSG00000075624" "ENSG00000184009" "ENSG00000142534" "ENSG00000177954" ...</span>
<span class="co">#&gt;   .. ..$ : chr [1:15] "R16-191_H5TKNBBXX" "R16-287_H5TKNBBXX" "R16-197_H5TKNBBXX" "R16-850_H7V3TBBXX" ...</span>
<span class="co">#&gt;  $ Geschwind.inVivo.sc.Hs  : num [1:40, 1:15] 9.52 11.32 9.52 10.51 8.52 ...</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   .. ..$ : chr [1:40] "ACTB" "ACTG1" "RPS11" "RPS27" ...</span>
<span class="co">#&gt;   .. ..$ : chr [1:15] "TGCTAATACTGA" "TTCACGATTTTT" "CTGTCAGAATAA" "CATAATATGTCA" ...</span>
<span class="co">#&gt;  $ Jabaudon.inVivo.sc.Mm   : num [1:40, 1:15] 11.55 11.98 10.03 8.85 9.82 ...</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   .. ..$ : chr [1:40] "Actb" "Actg1" "Rps11" "Rps27rt" ...</span>
<span class="co">#&gt;   .. ..$ : chr [1:15] "E14.24H_LTHT7_1_1" "E14.24H_LTHT7_1_2" "E14.24H_LTHT7_1_5" "E14.24H_LTHT7_1_6" ...</span></code></pre></div>
</div>
<div id="preprocess-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocess-data" class="anchor"></a>Preprocess data</h2>
<p>The four data sets contained in this demo are from either human or mouse species annotated with either gene symbol or ensembl IDs, we provide a wrapper function sjdwrap to help users preprocess and align datasets to be ready for decomposition.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SJDdataIN</span> <span class="op">=</span> <span class="fu"><a href="../reference/sjdWrap.html">sjdWrap</a></span><span class="op">(</span>
  data.list <span class="op">=</span> <span class="va">NeuroGenesis4</span>,
  species.vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human"</span>, <span class="st">"human"</span>, <span class="st">"human"</span>, <span class="st">"mouse"</span><span class="op">)</span>,
  geneType.vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"symbol"</span>,<span class="st">"ensembl"</span>,<span class="st">"symbol"</span>,<span class="st">"symbol"</span><span class="op">)</span>,
  geneType.out <span class="op">=</span> <span class="st">"symbol"</span>,
  species.out <span class="op">=</span> <span class="st">"human"</span>
  <span class="op">)</span>
<span class="co">#&gt; Using biomaRt to connect gene IDs across 4 datasets:</span>
<span class="co">#&gt; Getting biomaRt IDs for dataset 1</span>
<span class="co">#&gt; Ensembl site unresponsive, trying uswest mirror</span>
<span class="co">#&gt; You have input  40  genes</span>
<span class="co">#&gt; We found  40  matches</span>
<span class="co">#&gt; 1  of those are duplicates and only keeping the 1st of each</span>
<span class="co">#&gt; Getting biomaRt IDs for dataset 2</span>
<span class="co">#&gt; Ensembl site unresponsive, trying uswest mirror</span>
<span class="co">#&gt; You have input  40  genes</span>
<span class="co">#&gt; We found  37  matches</span>
<span class="co">#&gt; 1  of those are duplicates and only keeping the 1st of each</span>
<span class="co">#&gt; Getting biomaRt IDs for dataset 3 </span>
<span class="co">#&gt; You have input  40  genes</span>
<span class="co">#&gt; We found  39  matches</span>
<span class="co">#&gt; 0  of those are duplicates and only keeping the 1st of each</span>
<span class="co">#&gt; Getting biomaRt IDs for dataset 4</span>
<span class="co">#&gt; Ensembl site unresponsive, trying uswest mirror</span>
<span class="co">#&gt; You have input  40  genes</span>
<span class="co">#&gt; We found  34  matches</span>
<span class="co">#&gt; 0  of those are duplicates and only keeping the 1st of each</span>
<span class="co">#&gt; constructed 4 tables of cross-species matching genes</span>
<span class="co">#&gt; we found 27 shared genes in 4 datasets</span>
<span class="co">#&gt; new data list of 4 datasets constructed</span></code></pre></div>
</div>
<div id="apply-the-decomposition-method" class="section level2">
<h2 class="hasAnchor">
<a href="#apply-the-decomposition-method" class="anchor"></a>Apply the decomposition method</h2>
<p>After the transformation and alignment among the data sets, the next step is to apply the decomposition method, we give examples on different methods to illustrate the process</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  Shared.All.4 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">:</span> <span class="fl">4</span><span class="op">)</span>, 
  Shared.bulk.2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,
  Shared.sc.2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>,
  Hs.Meisnr.1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,
  Hs.AZ.1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,
  Gesch.1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>,
  Telley.1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>

<span class="va">dims</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span> <span class="co"># must have same length as "grp"</span>

<span class="co"># sep</span>
<span class="va">sepPCA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/sepPCA.html">sepPCA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">sepICA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/sepICA.html">sepICA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">sepNMF.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/sepNMF.html">sepNMF</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">dims</span><span class="op">)</span>

<span class="co"># concat</span>
<span class="va">concatPCA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/concatPCA.html">concatPCA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">concatICA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/concatICA.html">concatICA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">concatNMF.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/concatNMF.html">concatNMF</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>

<span class="co"># joint</span>
<span class="va">jointPCA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/jointPCA.html">jointPCA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">jointICA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/jointICA.html">jointICA</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>
<span class="va">jointNMF.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/jointNMF.html">jointNMF</a></span><span class="op">(</span><span class="va">SJDdataIN</span>, <span class="va">grp</span>, <span class="va">dims</span><span class="op">)</span>

<span class="co"># sequential</span>
<span class="va">twoStageLCA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/twoStageLCA.html">twoStageLCA</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">SJDdataIN</span>, group <span class="op">=</span> <span class="va">grp</span>, comp_num <span class="op">=</span> <span class="va">dims</span><span class="op">)</span>
<span class="va">twoStageiLCA.out</span> <span class="op">=</span> <span class="fu"><a href="../reference/twoStageiLCA.html">twoStageiLCA</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">SJDdataIN</span>, group <span class="op">=</span> <span class="va">grp</span>, comp_num <span class="op">=</span> <span class="va">dims</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot" class="section level2">
<h2 class="hasAnchor">
<a href="#plot" class="anchor"></a>Plot</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'gridExtra'</span>
<span class="co">#&gt; The following object is masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span></code></pre></div>
<div id="plot-each-dim-in-each-dataset-individually" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-each-dim-in-each-dataset-individually" class="anchor"></a>Plot each dim in each dataset individually</h3>
<p>Here we need to call the <code>SJDScorePlotter</code> function to generate the <code>SJDScorePlotter.obj</code>, a list of ggplot objects that can be used</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SampleMetaNamesTable</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">NeuroGenesis4</span><span class="op">)</span>,
    Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Yaxis'</span>,<span class="st">'Yaxis'</span>,<span class="st">'2Dscatter'</span>,<span class="st">'2Dscatter'</span><span class="op">)</span>,
    XaxisColumn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"X"</span>,<span class="st">"DAYx"</span>,<span class="st">"tSNE_1"</span>,<span class="st">"tsne1:ch1"</span><span class="op">)</span>,
    YaxisColumn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PJDscores"</span>,<span class="st">"PJDscores"</span>,<span class="st">"tSNE_2"</span>,<span class="st">"tsne2:ch1"</span><span class="op">)</span>,
    COLaxisColumn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"color"</span>,<span class="st">"colorBYlabelsX"</span>,<span class="st">"PJDscores"</span>,<span class="st">"PJDscores"</span><span class="op">)</span>,
    PCHColumn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">""</span>,<span class="st">""</span>,<span class="st">""</span>,<span class="st">""</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SJDScorePlotter.obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/SJDScorePlotter.html">SJDScorePlotter</a></span><span class="op">(</span>
     SJDalg <span class="op">=</span> <span class="st">"twoStageLCA"</span>,
     scores <span class="op">=</span> <span class="va">twoStageLCA.out</span><span class="op">$</span><span class="va">score_list</span>,
     lbb <span class="op">=</span> <span class="st">"NeuroGenesis4.p2"</span>,
     info <span class="op">=</span> <span class="va">NeuroGenesis4.info</span>,
     SampleMetaNamesTable <span class="op">=</span> <span class="va">SampleMetaNamesTable</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "**********************************************************"</span>
<span class="co">#&gt; [1] "plotting scores for twoStageLCA anaysis."</span>
<span class="co">#&gt; [1] "*******************************"</span>
<span class="co">#&gt; [1] "dataset: Meissner.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Shared.All.4"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.bulk.2"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.sc.2"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Shared.sc.2, moving to the next SJD output for Meissner.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Hs.Meisnr.1"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Hs.AZ.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.AZ.1, moving to the next SJD output for Meissner.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Gesch.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Gesch.1, moving to the next SJD output for Meissner.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Telley.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Telley.1, moving to the next SJD output for Meissner.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "*******************************"</span>
<span class="co">#&gt; [1] "dataset: LIBD.AZ.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Shared.All.4"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.bulk.2"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.sc.2"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Shared.sc.2, moving to the next SJD output for LIBD.AZ.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Hs.Meisnr.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.Meisnr.1, moving to the next SJD output for LIBD.AZ.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Hs.AZ.1"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Gesch.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Gesch.1, moving to the next SJD output for LIBD.AZ.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "grouping: Telley.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Telley.1, moving to the next SJD output for LIBD.AZ.inVitro.bulk.Hs"</span>
<span class="co">#&gt; [1] "*******************************"</span>
<span class="co">#&gt; [1] "dataset: Geschwind.inVivo.sc.Hs"</span>
<span class="co">#&gt; [1] "grouping: Shared.All.4"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.bulk.2"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Shared.bulk.2, moving to the next SJD output for Geschwind.inVivo.sc.Hs"</span>
<span class="co">#&gt; [1] "grouping: Shared.sc.2"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Hs.Meisnr.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.Meisnr.1, moving to the next SJD output for Geschwind.inVivo.sc.Hs"</span>
<span class="co">#&gt; [1] "grouping: Hs.AZ.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.AZ.1, moving to the next SJD output for Geschwind.inVivo.sc.Hs"</span>
<span class="co">#&gt; [1] "grouping: Gesch.1"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Telley.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Telley.1, moving to the next SJD output for Geschwind.inVivo.sc.Hs"</span>
<span class="co">#&gt; [1] "*******************************"</span>
<span class="co">#&gt; [1] "dataset: Jabaudon.inVivo.sc.Mm"</span>
<span class="co">#&gt; [1] "grouping: Shared.All.4"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Shared.bulk.2"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Shared.bulk.2, moving to the next SJD output for Jabaudon.inVivo.sc.Mm"</span>
<span class="co">#&gt; [1] "grouping: Shared.sc.2"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] "grouping: Hs.Meisnr.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.Meisnr.1, moving to the next SJD output for Jabaudon.inVivo.sc.Mm"</span>
<span class="co">#&gt; [1] "grouping: Hs.AZ.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Hs.AZ.1, moving to the next SJD output for Jabaudon.inVivo.sc.Mm"</span>
<span class="co">#&gt; [1] "grouping: Gesch.1"</span>
<span class="co">#&gt; [1] "No SJD scores for grouping : Gesch.1, moving to the next SJD output for Jabaudon.inVivo.sc.Mm"</span>
<span class="co">#&gt; [1] "grouping: Telley.1"</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; [1] 2</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">SJDScorePlotter.obj</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  [1] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp1of2" </span>
<span class="co">#&gt;  [2] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp2of2" </span>
<span class="co">#&gt;  [3] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_Meissner.inVitro.bulk.Hs.comp1of2"</span>
<span class="co">#&gt;  [4] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_Meissner.inVitro.bulk.Hs.comp2of2"</span>
<span class="co">#&gt;  [5] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.Meisnr.1.data_Meissner.inVitro.bulk.Hs.comp1of2"  </span>
<span class="co">#&gt;  [6] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.Meisnr.1.data_Meissner.inVitro.bulk.Hs.comp2of2"  </span>
<span class="co">#&gt;  [7] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_LIBD.AZ.inVitro.bulk.Hs.comp1of2"  </span>
<span class="co">#&gt;  [8] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_LIBD.AZ.inVitro.bulk.Hs.comp2of2"  </span>
<span class="co">#&gt;  [9] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_LIBD.AZ.inVitro.bulk.Hs.comp1of2" </span>
<span class="co">#&gt; [10] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_LIBD.AZ.inVitro.bulk.Hs.comp2of2" </span>
<span class="co">#&gt; [11] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.AZ.1.data_LIBD.AZ.inVitro.bulk.Hs.comp1of2"       </span>
<span class="co">#&gt; [12] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.AZ.1.data_LIBD.AZ.inVitro.bulk.Hs.comp2of2"       </span>
<span class="co">#&gt; [13] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Geschwind.inVivo.sc.Hs.comp1of2"   </span>
<span class="co">#&gt; [14] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Geschwind.inVivo.sc.Hs.comp2of2"   </span>
<span class="co">#&gt; [15] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.sc.2.data_Geschwind.inVivo.sc.Hs.comp1of2"    </span>
<span class="co">#&gt; [16] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.sc.2.data_Geschwind.inVivo.sc.Hs.comp2of2"    </span>
<span class="co">#&gt; [17] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Gesch.1.data_Geschwind.inVivo.sc.Hs.comp1of2"        </span>
<span class="co">#&gt; [18] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Gesch.1.data_Geschwind.inVivo.sc.Hs.comp2of2"        </span>
<span class="co">#&gt; [19] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Jabaudon.inVivo.sc.Mm.comp1of2"    </span>
<span class="co">#&gt; [20] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Jabaudon.inVivo.sc.Mm.comp2of2"    </span>
<span class="co">#&gt; [21] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.sc.2.data_Jabaudon.inVivo.sc.Mm.comp1of2"     </span>
<span class="co">#&gt; [22] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.sc.2.data_Jabaudon.inVivo.sc.Mm.comp2of2"     </span>
<span class="co">#&gt; [23] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Telley.1.data_Jabaudon.inVivo.sc.Mm.comp1of2"        </span>
<span class="co">#&gt; [24] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Telley.1.data_Jabaudon.inVivo.sc.Mm.comp2of2"</span></code></pre></div>
<p>Save the whole list to a .RData file</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># save(SJDScorePlotter.obj, file = "./SJDScorePlotter.obj.RData")</span></code></pre></div>
</div>
<div id="assemble-the-images-based-on-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#assemble-the-images-based-on-dataset" class="anchor"></a>Assemble the images based on dataset</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">assemble.byDataset.obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/assemble.byDataset.html">assemble.byDataset</a></span><span class="op">(</span>SJDScorePlotter.obj <span class="op">=</span> <span class="va">SJDScorePlotter.obj</span>, dataset_name <span class="op">=</span> <span class="st">"Meissner.inVitro.bulk.Hs"</span>, SJD_algorithm <span class="op">=</span> <span class="st">"twoStageLCA"</span>, group <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">assemble.byDataset.obj</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp1of2" </span>
<span class="co">#&gt; [2] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp2of2" </span>
<span class="co">#&gt; [3] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_Meissner.inVitro.bulk.Hs.comp1of2"</span>
<span class="co">#&gt; [4] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.bulk.2.data_Meissner.inVitro.bulk.Hs.comp2of2"</span>
<span class="co">#&gt; [5] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.Meisnr.1.data_Meissner.inVitro.bulk.Hs.comp1of2"  </span>
<span class="co">#&gt; [6] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Hs.Meisnr.1.data_Meissner.inVitro.bulk.Hs.comp2of2"</span></code></pre></div>
<p>Plot all images corresponding from the <code>assemble.byDataset.obj</code>. We can construct them into a whole big image through the <code>grid.arrange</code> from the <code>grid.Extra</code> package.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span>
  <span class="va">assemble.byDataset.obj</span><span class="op">$</span><span class="va">SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp1of2</span>,
  <span class="va">assemble.byDataset.obj</span><span class="op">$</span><span class="va">SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp2of2</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>, ncol <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span></code></pre></div>
<p><img src="SJD_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="assemble-the-images-based-on-component" class="section level3">
<h3 class="hasAnchor">
<a href="#assemble-the-images-based-on-component" class="anchor"></a>Assemble the images based on component</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">assemble.byComponent.obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/assemblePNG.byComponent.html">assemblePNG.byComponent</a></span><span class="op">(</span>SJDScorePlotter.obj <span class="op">=</span> <span class="va">SJDScorePlotter.obj</span>, component <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, SJD_algorithm <span class="op">=</span> <span class="st">"twoStageLCA"</span>, group <span class="op">=</span> <span class="st">'Shared.All.4'</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">assemble.byComponent.obj</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp1of2"</span>
<span class="co">#&gt; [2] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_LIBD.AZ.inVitro.bulk.Hs.comp1of2" </span>
<span class="co">#&gt; [3] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Geschwind.inVivo.sc.Hs.comp1of2"  </span>
<span class="co">#&gt; [4] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Jabaudon.inVivo.sc.Mm.comp1of2"   </span>
<span class="co">#&gt; [5] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Meissner.inVitro.bulk.Hs.comp2of2"</span>
<span class="co">#&gt; [6] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_LIBD.AZ.inVitro.bulk.Hs.comp2of2" </span>
<span class="co">#&gt; [7] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Geschwind.inVivo.sc.Hs.comp2of2"  </span>
<span class="co">#&gt; [8] "SJDout_NeuroGenesis4.p2.SJDalg_twoStageLCA.grp_Shared.All.4.data_Jabaudon.inVivo.sc.Mm.comp2of2"</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span>
  <span class="va">assemble.byComponent.obj</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  <span class="va">assemble.byComponent.obj</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>, ncol <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span></code></pre></div>
<p><img src="SJD_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
</div>
<div id="reference" class="section level1">
<h1 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h1>
<ul>
<li>Chen, H., Caffo, B., Stein-O’Brien, G., Liu, J., Langmead, B., Colantuoni, C., &amp; Xiao, L. (2021). Two-stage Linked Component Analysis for Joint Decomposition of Multiple Biologically Related Data Sets. bioRxiv.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Huan Chen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
