<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="A list of info for 4 neuro data sets sampled, containing: Meissner.inVitro.bulk.Hs, LIBD.AZ.inVitro.bulk.Hs, Geschwind.inVivo.sc.Hs, Jabaudon.inVivo.sc.Mm"><title>NeuroGenesis4.info — NeuroGenesis4.info • SJD</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="NeuroGenesis4.info — NeuroGenesis4.info"><meta property="og:description" content="A list of info for 4 neuro data sets sampled, containing: Meissner.inVitro.bulk.Hs, LIBD.AZ.inVitro.bulk.Hs, Geschwind.inVivo.sc.Hs, Jabaudon.inVivo.sc.Mm"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SJD</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
    Home
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/SJD.html">Demo</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-real-data-analysis">Real Data Analysis</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-real-data-analysis">
    <a class="dropdown-item" href="../articles/StructralJointDecomposition.html">Introduction to SJD (Structured Joint Decomposition)</a>
    <a class="dropdown-item" href="../articles/twoStageLCA.html">Introduction to twoStageLCA (Two Stage Linked Component Analysis)</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/CHuanSite/SJD/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>NeuroGenesis4.info</h1>
      <small class="dont-index">Source: <a href="https://github.com/CHuanSite/SJD/blob/HEAD/R/neuroGenesis4.R" class="external-link"><code>R/neuroGenesis4.R</code></a></small>
      <div class="d-none name"><code>NeuroGenesis4.info.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A list of info for 4 neuro data sets sampled, containing: Meissner.inVitro.bulk.Hs, LIBD.AZ.inVitro.bulk.Hs, Geschwind.inVivo.sc.Hs, Jabaudon.inVivo.sc.Mm</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NeuroGenesis4</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A list containing 4 dataframes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NeuroGenesis4.info</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NeuroGenesis4.info</span><span class="op">$</span><span class="va">Meissner.inVitro.bulk.Hs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         IDs                     File Cond Rep X      color</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ESC.1 ESC.1 hESC_rep1.count.fpkm.txt  ESC   1 1      black</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ESC.2 ESC.2 hESC_rep2.count.fpkm.txt  ESC   2 1      black</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NE.1   NE.1   NE_rep1.count.fpkm.txt   NE   1 2 lightpink2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NE.2   NE.2   NE_rep2.count.fpkm.txt   NE   2 2 lightpink2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ERG.1 ERG.1  ERG_rep1.count.fpkm.txt  ERG   1 3 indianred1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ERG.2 ERG.2  ERG_rep2.count.fpkm.txt  ERG   2 3 indianred1</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NeuroGenesis4.info</span><span class="op">$</span><span class="va">LIBD.AZ.inVitro.bulk.Hs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     EXP_STATUS RNAseq_STATUS Order  RNA_NO     SAMPLE_NAME EXPERIMENT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126   Complete      Complete   126 R16-191    165-B-2 D2 1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164   Complete      Complete   164 R16-287  165-B-2 D9 2-1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127   Complete      Complete   127 R16-197    165-B-3 D2 1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469   Complete      Complete    NA R16-850 165-B-3_D2_REP1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470   Complete      Complete    NA R16-851 165-B-3_D6_REP1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165   Complete      Complete   165 R16-288  165-B-3 D9 2-1 TIMECOURSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     RNA_CONC_ng/ul Library LibConstruct   DateToAJ SPECIES      JC_Comment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126             NA  Batch4   2016-03-31 2016-04-13   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164             NA  Batch4   2016-03-31 2016-04-13   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127             NA  Batch4   2016-03-31 2016-04-13   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469          423.4 BATCH10   2016-09-08 2016-09-28   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470          251.5 BATCH10   2016-09-08 2016-09-28   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165             NA  Batch4   2016-03-31 2016-04-13   HUMAN ASSAY_1.1_COND2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Flowcell    Index Lane  NumReads            SeqNotes NumMapped   mapRate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126 H5TKNBBXX ATTCAGAA   L8  70256642                &lt;NA&gt;  64493777 0.9179741</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164 H5TKNBBXX CGGCTATG   L2  47653100                &lt;NA&gt;  38805346 0.8143299</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127 H5TKNBBXX AGCGATAG   L4  84345206 10.64% undetermined  79142539 0.9383170</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469 H7V3TBBXX GAGATTCC   L6 143106660                &lt;NA&gt; 133682681 0.9341472</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470 H7V3TBBXX GAATTCGT   L6 174230786                &lt;NA&gt; 161315252 0.9258711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165 H5TKNBBXX TCCGGAGA   L1  76574342                &lt;NA&gt;  66510301 0.8685716</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     NumProperMap properRate NumUnmapped unmapRate riboMapped    riboRate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126     44317316  0.6871565    10866616 0.1546703    1848394 0.028660036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164     27965698  0.7206661    12397154 0.2601542    2141647 0.055189483</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127     54477558  0.6883474    11344493 0.1345007     124652 0.001575032</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469     98673366  0.7381163    17993507 0.1257349    2816740 0.021070343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470    122875468  0.7617102    23249854 0.1334429    1977016 0.012255605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165     44773122  0.6731758    15654357 0.2044335    3235748 0.048650329</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mitoMapped    mitoRate alignNote         Class Donor    LINE  DX RPS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126     951371 0.014924190      &lt;NA&gt; Naked genomes   165 165-B-2 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164     523056 0.013734300      &lt;NA&gt; Naked genomes   165 165-B-2 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127     434379 0.005578326      &lt;NA&gt; Naked genomes   165 165-B-3 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469     935448 0.007175070      &lt;NA&gt; Naked genomes   165 165-B-3 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470     873678 0.005559844      &lt;NA&gt; Naked genomes   165 165-B-3 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165     796443 0.012205907      &lt;NA&gt; Naked genomes   165 165-B-3 CNT  lo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Manipulation shRNA MOI PURO   SAMPLE_NAME.1 shRNA_Target_Seq Note DAY</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126           NO    NA  NA   NA    165-B-2 D2 1               NA &lt;NA&gt;   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164           NO    NA  NA   NA  165-B-2 D9 2-1               NA &lt;NA&gt;   9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127           NO    NA  NA   NA    165-B-3 D2 1               NA &lt;NA&gt;   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469           NO    NA  NA   NA 165-B-3_D2_REP1               NA &lt;NA&gt;   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470           NO    NA  NA   NA 165-B-3_D6_REP1               NA &lt;NA&gt;   6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165           NO    NA  NA   NA  165-B-3 D9 2-1               NA &lt;NA&gt;   9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BIO_REP Fluidigm     CONDITION          SampleID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126       1       NA ACC_DORSAL(2) R16-191_H5TKNBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164       1       NA ACC_DORSAL(2) R16-287_H5TKNBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127       1       NA ACC_DORSAL(2) R16-197_H5TKNBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469       1       NA ACC_DORSAL(2) R16-850_H7V3TBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470       1       NA ACC_DORSAL(2) R16-851_H7V3TBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165       1       NA ACC_DORSAL(2) R16-288_H5TKNBBXX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                    bamFile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-191_H5TKNBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-287_H5TKNBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-197_H5TKNBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-850_H7V3TBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-851_H7V3TBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165 /dcl01/lieber/ajaffe/Astrazeneca/TopHat/Sample_R16-288_H5TKNBBXX_out/accepted_hits.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     leftInput leftMapped leftMapRate rightInput rightMapped rightMapRate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126  35128321   30441590       0.867   35128321    28948436        0.824</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164  23826550   17892853       0.751   23826550    17363093        0.729</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127  42172603   37200746       0.882   42172603    35799967        0.849</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469  71553330   63377397       0.886   71553330    61735756        0.863</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470  87115393   76260852       0.875   87115393    74720080        0.858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165  38287171   31423079       0.821   38287171    29496906        0.770</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overallMapRate concordMapRate totalMapped totalAssignedGene lineRep</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126          0.845          0.782    62795541         0.6497077       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164          0.740          0.669    37560864         0.6179368       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127          0.865          0.806    77434676         0.7626106       3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469          0.874          0.825   129439310         0.7630018       3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470          0.867          0.824   156267057         0.7672984       3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165          0.796          0.715    64454176         0.6493536       3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     colDonor PROG.21-B-8 PROG.165-B-6X PROG.66-A-3 PROG.90-A-5 PROG.21-B-9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165    black       FALSE         FALSE       FALSE       FALSE       FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     PROG.165-B-8X PROG.66-A-9 PROG.90-A-10 PROG.165-B-2 PROG.165-B-3 PROG.3-A-7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126         FALSE       FALSE        FALSE         TRUE        FALSE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164         FALSE       FALSE        FALSE         TRUE        FALSE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127         FALSE       FALSE        FALSE        FALSE         TRUE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469         FALSE       FALSE        FALSE        FALSE         TRUE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470         FALSE       FALSE        FALSE        FALSE         TRUE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165         FALSE       FALSE        FALSE        FALSE         TRUE      FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     PROG.3-A-8 PROG.165-B-3X PROG.21-B-3  DxRps pchBYdiagRPS     DAYj DAYx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126      FALSE         FALSE       FALSE CNT.lo           19 1.431041    4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164      FALSE         FALSE       FALSE CNT.lo           19 8.883865    7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127      FALSE         FALSE       FALSE CNT.lo           19 1.092173    4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469      FALSE         FALSE       FALSE CNT.lo           19 2.241799    4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470      FALSE         FALSE       FALSE CNT.lo           19 6.932188    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165      FALSE         FALSE       FALSE CNT.lo           19 9.986775    7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     labelsX colorBYlabelsX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 126       2            red</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 164       9            red</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 127       2            red</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 469       2            red</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 470       6            red</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 165       9            red</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NeuroGenesis4.info</span><span class="op">$</span><span class="va">Geschwind.inVivo.sc.Hs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Cell Cluster Subcluster Donor Layer Gestation_week Index Library</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 169 TGCTAATACTGA     vRG      vRG_0   368    CP             17  N701   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 624 TTCACGATTTTT     vRG      vRG_2   368    GZ             17  N702   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 659 CTGTCAGAATAA     vRG      vRG_2   368    GZ             17  N702   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 681 CATAATATGTCA     vRG      vRG_0   368    GZ             17  N702   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 715 TGCCAATCCGTT     vRG      vRG_0   368    GZ             17  N702   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 723 TCGACTATTTTG     vRG      vRG_2   368    GZ             17  N702   Plath</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Number_genes_detected Number_UMI Percentage_mitochondrial S_phase_score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 169                  1366       2736                     4.20         -0.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 624                  3012       7318                     1.87          0.83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 659                  1597       3451                     2.17         -0.13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 681                  1551       2941                     1.43          0.38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 715                  1569       2769                     1.81         -0.04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 723                  1610       2994                     3.57          0.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     G2M_phase_score Phase            X    tSNE_1   tSNE_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 169          -0.073    G1 TGCTAATACTGA -3.446397 17.68239</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 624           0.350     S TTCACGATTTTT -8.190806 22.31286</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 659          -0.110    G1 CTGTCAGAATAA -3.301333 18.49425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 681           0.064     S CATAATATGTCA -8.502828 16.35404</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 715          -0.170    G1 TGCCAATCCGTT -3.293814 19.02176</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 723           0.064     S TCGACTATTTTG -7.173418 21.53491</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NeuroGenesis4.info</span><span class="op">$</span><span class="va">Jabaudon.inVivo.sc.Mm</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         title geo_accession                status</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837  E14.24H_LTHT7_1_1    GSM3351837 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838  E14.24H_LTHT7_1_2    GSM3351838 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839  E14.24H_LTHT7_1_5    GSM3351839 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840  E14.24H_LTHT7_1_6    GSM3351840 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841  E14.24H_LTHT7_1_7    GSM3351841 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 E14.24H_LTHT7_1_12    GSM3351842 Public on May 10 2019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            submission_date last_update_date type channel_count source_name_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842     Aug 23 2018      May 10 2019  SRA             1  cortical cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            organism_ch1 characteristics_ch1 characteristics_ch1.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 Mus musculus       tsne1: -21.73          tsne2: 10.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 Mus musculus       tsne1: -12.66          tsne2: -0.39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 Mus musculus         tsne1: 4.78           tsne2: 2.06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 Mus musculus       tsne1: -14.95           tsne2: 7.12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 Mus musculus       tsne1: -15.24           tsne2: 7.04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 Mus musculus       tsne1: -19.39          tsne2: 11.40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.2 characteristics_ch1.3 characteristics_ch1.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837      num_read: 608677          pct_mt: 4.3%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838       num_read: 68123          pct_mt: 4.8%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839      num_read: 608394          pct_mt: 3.5%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840      num_read: 557976          pct_mt: 3.4%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841      num_read: 121333          pct_mt: 8.7%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842      num_read: 621786          pct_mt: 2.9%     cluster: N1d.late</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                characteristics_ch1.5 characteristics_ch1.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 pup.age.at.injection: E14         cell.age: 24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    characteristics_ch1.7      characteristics_ch1.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 pseudo.differentiation: 0.409 pseudo.birthdate24h: 0.417</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 pseudo.differentiation: 0.381 pseudo.birthdate24h: 0.462</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 pseudo.differentiation: 0.327 pseudo.birthdate24h: 0.459</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 pseudo.differentiation: 0.378 pseudo.birthdate24h: 0.505</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 pseudo.differentiation: 0.372 pseudo.birthdate24h: 0.500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 pseudo.differentiation: 0.421 pseudo.birthdate24h: 0.431</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   characteristics_ch1.9             characteristics_ch1.10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 pseudo.birthdate.pred: 0.333 pseudo.differentiation.pred: 0.592</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 pseudo.birthdate.pred: 0.152 pseudo.differentiation.pred: 0.130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 pseudo.birthdate.pred: 0.890 pseudo.differentiation.pred: 0.134</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 pseudo.birthdate.pred: 0.439 pseudo.differentiation.pred: 0.119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 pseudo.birthdate.pred: 0.049 pseudo.differentiation.pred: 0.056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 pseudo.birthdate.pred: 0.578 pseudo.differentiation.pred: 0.172</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      characteristics_ch1.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 pseudo.birthdate.adj.pred: 3.411</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 pseudo.birthdate.adj.pred: 3.434</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 pseudo.birthdate.adj.pred: 3.963</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 pseudo.birthdate.adj.pred: 3.316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 pseudo.birthdate.adj.pred: 2.687</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 pseudo.birthdate.adj.pred: 3.553</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            characteristics_ch1.12 molecule_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 pseudo.differentiation.adj.pred: 2.219    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 pseudo.differentiation.adj.pred: 1.982    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 pseudo.differentiation.adj.pred: 1.982    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 pseudo.differentiation.adj.pred: 1.943    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 pseudo.differentiation.adj.pred: 1.919    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 pseudo.differentiation.adj.pred: 1.908    polyA RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        extract_protocol_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 The putative primary somatosensory cortex S1 was microdissected and incubated in 0.05% trypsin at 37°C for 5 minutes. Following tissue digestion, fetal bovine serum was added to the mix and cells were manually dissociated via up-and-down pipetting. Cells were centrifuged 5min at 300g and the pellet was suspended in 1ml of HBSS then passed on a 70µm cell stainer. FT+ cells, gated on the top 5% brightest cells, were finally FAC-sorted on a MoFloAstrios (Beckman). FAC-sorted FT+ cells (18µl) were mixed with the C1 Suspension Reagent (2µl; Fluidigm) yielding a total of 20µl of cell suspension mix with ~500 cells/μl. The cell suspension mix was loaded on a C1 Single-Cell AutoPrep integrated fluidic circuit (IFC) designed for 10- to 17-µm cells (HT-800, Fluidigm #100-57-80). cDNA synthesis and preamplification was processed following the manufacturer’s instructions (C1 system, Fluidigm). Single cell RNA-sequencing libraries of the cDNA were prepared using Nextera XT DNA library prep kit (Illumina). Libraries were multi-plexed and sequenced according to the manufacturer’s recommendations with paired-end reads using HiSeq2500 platform (Illumina) with an expected depth of 1M reads per single cell, and a final mapping read length of 70bp.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                          extract_protocol_ch1.1 taxid_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 SMARTseq v4 kit (Clontech, # 634888)     10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                   data_processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 The 5bp UMI sequence (initially sequenced in read1), are appended at the end of the headers of the reads (here the reads we provide are resulting from this operation)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                  data_processing.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 reads are mapped with tophat v2.0.13 on the mouse genome GRCm38 considering Ensembl78 genes annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                  data_processing.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 bam files are processed with umi_tools to deduplicate reads with identical UMI sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                           data_processing.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 gene expression are quantifyied using summarizeOverlaps() method of the R package GenomicsAlignments(). Only exonic reads are considered in the quantification and this includes 5' and 3' UTRs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               data_processing.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 Genome_build: GRCm38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                  data_processing.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 Supplementary_files_format_and_content: raw_count.tsv.gz contains the result of gene expression quantification by summarizeOverlaps() for each cell. The integers values represent the number of exonic reads mapping into each gene after UMI correction. The first columns contains the gene name that is quantified.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            platform_id   contact_name          contact_email    contact_phone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842    GPL17021 Julien,,Prados julien.prados@unige.ch +41 22 37 95 396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_laboratory contact_department    contact_institute</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842     Denis Jabaudon      Neurosciences University of Geneva</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                contact_address contact_city contact_zip/postal_code</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 Rue Michel Servet 1       Geneva                    1211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_country data_row_count    instrument_model library_selection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842     Switzerland              0 Illumina HiSeq 2500              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            library_source library_strategy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 transcriptomic          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                  relation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901434</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901431</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901430</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN09901429</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                       relation.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600063</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600065</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600067</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX4600068</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            supplementary_file_1 cell.age:ch1 cluster:ch1 num_read:ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837                 NONE          24H    N1d.late       608677</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838                 NONE          24H    N1d.late        68123</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839                 NONE          24H    N1d.late       608394</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840                 NONE          24H    N1d.late       557976</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841                 NONE          24H    N1d.late       121333</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842                 NONE          24H    N1d.late       621786</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            pct_mt:ch1 pseudo.birthdate.adj.pred:ch1 pseudo.birthdate.pred:ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837       4.3%                         3.411                     0.333</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838       4.8%                         3.434                     0.152</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839       3.5%                         3.963                     0.890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840       3.4%                         3.316                     0.439</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841       8.7%                         2.687                     0.049</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842       2.9%                         3.553                     0.578</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            pseudo.birthdate24h:ch1 pseudo.birthdate96h:ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837                   0.417                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838                   0.462                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839                   0.459                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840                   0.505                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841                   0.500                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842                   0.431                    &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            pseudo.birthdateprog:ch1 pseudo.differentiation:ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837                     &lt;NA&gt;                      0.409</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838                     &lt;NA&gt;                      0.381</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839                     &lt;NA&gt;                      0.327</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840                     &lt;NA&gt;                      0.378</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841                     &lt;NA&gt;                      0.372</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842                     &lt;NA&gt;                      0.421</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            pseudo.differentiation.adj.pred:ch1 pseudo.differentiation.pred:ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837                               2.219                           0.592</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838                               1.982                           0.130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839                               1.982                           0.134</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840                               1.943                           0.119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841                               1.919                           0.056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842                               1.908                           0.172</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            pup.age.at.injection:ch1 tsne1:ch1 tsne2:ch1 pupAGEcellAGE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351837                      E14    -21.73     10.41       E14.24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351838                      E14    -12.66     -0.39       E14.24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351839                      E14      4.78      2.06       E14.24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351840                      E14    -14.95      7.12       E14.24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351841                      E14    -15.24      7.04       E14.24H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM3351842                      E14    -19.39     11.40       E14.24H</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Huan Chen, Jinrui Liu, Carlo Colantuoni.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

